scripts/files/fs.py

```diff
@@ -42,3 +49,62 @@ def exists(path: str) -> bool:
     if is_s3(path):
         return fs_s3.exists(path)
     return fs_local.exists(path)
+
+
+def write_all(inputs: List[str], target: str, concurrency: Optional[int] = 10) -> List[str]:
+    """Writes list of files to target destination using multithreading.
+
+    Args:
+        inputs: list of files to read
+        target: target folder to write to
+
+    Returns:
+        list of written file paths
+    """
+    written_tiffs: List[str] = []
+    with ThreadPoolExecutor(max_workers=concurrency) as executor:
+        futuress = {
+            executor.submit(write, os.path.join(target, f"{os.path.basename(input)}"), read(input)): input for input in inputs
```

Comment:
does this stream the file down or does it try and read all the files into memory and then write them out?

Reply:
After our discussion I have reduced the concurrecy to 4 for now, I also added a comment to [TDE-827](https://toitutewhenua.atlassian.net/browse/TDE-827) to work on this as a part of the review/refactor of our repos.